editions/tw5.com/tiddlers/widgets/TranscludeWidget.tid

```diff
@@ -1,24 +1,178 @@
+|`<$transclude $index="beta"/>` |Transcludes the index "beta" of the current tiddler |
+|`<$transclude $tiddler="foo" $index="beta"/>` |Transcludes the index "beta" of the tiddler "foo" |
+
+! Transclusion Parameters
+
+Named string parameters can be passed to the <<.wlink TranscludeWidget>> widget. They are made available as variables within the transcluded text. Parameters are only supported in modern mode.
+
+When invoking a transclusion, parameters are specified as additional attributes that do not start with a dollar sign `$`:
+
+```
+<$transclude $tiddler="MyTiddler" firstParameter="One" secondParameter="Two"/>
+```
+
+To pass parameters whose names start with a dollar sign `$`, prefix them with an extra `$`. For example, to pass a parameter called `$tiddler`:
+
+```
+<$transclude $tiddler="MyTiddler" $$tiddler="One"/>
+```
+
+There are several different ways to declare parameters within a transclusion:
+
+* the <<.wlink ParametersWidget>> widget
+* the [[Pragma: \parameters]]
+* the [[Pragma: \procedure]] for declaring procedure
+* the [[Pragma: \widgets]] for declaring custom widgets
+* the [[Pragma: \define]] for declaring macros
+
+An example of declaring parameters with the <<.wlink ParametersWidget>> widget:
+
+```
+<$parameters firstParameter="default" secondParameter="another default">
+    Parameters are available here as the variables <<firstParameter>> and <<secondParameter>>.
+</$parameters>
+```
+
+The `\parameters` [[Pragma: \parameters]] can be used as a shortcut syntax for declaring parameters. For example:
+
+```
+\parameters (firstParameter:"default",secondParameter:"another default")
+Parameters are available here as the variables <<firstParameter>> and <<secondParameter>>.
+```
+
+! Transclusion Slots
+
+Transcluded content can define special named locations called slots. At the point of transclusion, blocks of wikitext can be passed to the <<.wlink TranscludeWidget>> widget to fill those slots.
+
+Slots work very similarly to parameters except that they can contain structured wikitext, and not just plain text.
+
+For example, here we transclude the tiddler "Example" while using the <<.wlink FillWidget>> widget to pass wikitext blocks to fill the slots called "positive" and "negative":
+
+```
+<$transclude $tiddler="Example">
+  <$fill $name="positive">
+    <h1>This is positive</h1>
+  </$fill>
+  <$fill $name="negative">
+    <h3>This is negative</h3>
+  </$fill>
+</$transclude>
+```
+
+The tiddler "Example" uses the <<.wlink SlotWidget>> widget to specify the slots to be filled:
+
+```
+<ol>
+  <li><$slot $name="positive"/></li>
+  <li><$slot $name="negative"/></li>
+</ol>
+```
 
-The TranscludeWidget treats any contained content as a fallback if the target of the transclusion is not defined (ie a missing tiddler or a missing field).
+The output will be equivalent to:
+
+```
+<ol>
+  <li>
+    <h1>This is positive</h1>
+  </li>
+  <li>
+    <h3>This is negative</h3>
+  </li>
+</ol>
+```
```

Comment:
The above example does well to show the mechanics of how to use `$fill` and `$slot`, but it seems the same result can be achieved using parameters.

Tiddler 'Example 2':

```
\parameters (positive, negative)

<ol>
  <li><<positive>></li>
  <li><<negative>></li>
</ol>
```
Transclude the above using this:

```
<$transclude
  $tiddler="Example 2"
  positive="<h1>This is positive</h1>"
  negative="<h3>This is negative</h3>"
/>
```

It renders the same as the `$fill` and `$slot` example. Can the example be changed to give a better clue on when to choose fill/slot over parameters? Or give some explanatory guidance on when to use one over the other?

Reply:
Thanks @btheado the main advantage of slots over parameters is that quotes aren't needed. Even using triple double quotes for attributes isn't a universal fix because such quotes can't be nested. I'll update the docs to clarify.